src/v/cluster/data_migration_types.h

```diff
@@ -192,11 +214,15 @@ struct outbound_migration
 
     outbound_migration copy() const;
 
+    static std::optional<state> next_ntp_state(state state);
+
     auto serde_fields() { return std::tie(topics, groups, copy_to); }
 
     friend bool operator==(const outbound_migration&, const outbound_migration&)
       = default;
     friend std::ostream& operator<<(std::ostream&, const outbound_migration&);
+
+    auto topic_nts() const { return std::as_const(topics) | std::views::all; }
```

Comment:
why do we need a `view` here ? would returning `topics` const ref be enough ?

Reply:
The one for inbound migrations returns a view for a reason. If I made them so different the caller would need to make certain arrangements for it. E.g. for a view we need to use `ss::do_with` to make sure it survives the call, but for a const ref we cannot use `ss::do_with`, as it would attempt to copy topics. Ideally it should be solved with `ssx::smart_do_with_use_with_care_you_have_been_warned`, which would only hold rvalues passed and trust the caller to care about the lvalue references.